internal/service/finspace/kx_cluster_test.go

```diff
@@ -269,6 +337,140 @@ func TestAccFinSpaceKxCluster_rdb(t *testing.T) {
+			},
+			{
+				Config:      testAccKxClusterConfig_rdbEnvironmentId(rName, invalidEnvironmentId),
+				ExpectError: regexp.MustCompile("ResourceNotFoundException: No matching Environment found with requested ID"),
+			},
+		},
+	})
+}
+
+func TestAccFinSpaceKxCluster_rdbInvalidDatabaseName(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	ctx := acctest.Context(t)
+	var kxEnvironment finspace.GetKxEnvironmentOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	databaseName := "invalid-db"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckKxClusterDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKxClusterConfigBase(rName),
+				Check:  testAccCheckKxEnvironmentExists(ctx, "aws_finspace_kx_environment.test", &kxEnvironment),
+			},
+			{
+				Config:      testAccKxClusterConfig_rdbDatabaseName(rName, databaseName),
+				ExpectError: regexp.MustCompile("ResourceNotFoundException: Kx database"),
+			},
+		},
+	})
+}
+
+func TestAccFinSpaceKxCluster_rdbSaveDown(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	ctx := acctest.Context(t)
+	var kxEnvironment finspace.GetKxEnvironmentOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	saveDownType := "SDS01"
+	saveDownSize := "10"
+	emptySaveDownType := ""
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckKxClusterDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKxClusterConfigBase(rName),
+				Check:  testAccCheckKxEnvironmentExists(ctx, "aws_finspace_kx_environment.test", &kxEnvironment),
+			},
+			{
+				Config:      testAccKxClusterConfig_rdbSaveDown(rName, emptySaveDownType, saveDownSize),
+				ExpectError: regexp.MustCompile("Error: expected savedown_storage_configuration.0.type to be one of"),
+			},
+			{
+				Config:      testAccKxClusterConfig_rdbSaveDownNoSize(rName, saveDownType),
+				ExpectError: regexp.MustCompile("Error: Missing required argument"),
+			},
+			{
+				Config:      testAccKxClusterConfig_rdbNoSaveDownConfig(rName),
+				ExpectError: regexp.MustCompile("ValidationException: A cluster of type RDB must have savedown storage specified"),
+			},
+		},
+	})
+}
+
+func TestAccFinSpaceKxCluster_rdbCacheNoSaveDown(t *testing.T) {
```

Comment:
Just curious, do we know why we specifically want to test the savedown storage check both for an rdb with caching and without? If it's just to match existing integ tests that's all good, but just curious for the reasoning behind it

Reply:
It's solely to match the existing integration test scenarios